sparkmagic/sparkmagic/livyclientlib/reliablehttpclient.py

```diff
@@ -78,6 +80,8 @@ def _send_request_helper(self, url, accepted_status_codes, function, data, retry
                 error = False
                 status = r.status_code
                 text = r.text
+                if self._endpoint.auth == constants.AUTH_BASIC and self._endpoint.password:
+                    self._endpoint.password = ""
```

Comment:
Setting the password field to empty string is not the way to achieve this. Reading several threads online about python security, I'm not sure it's that achievable either...

So please remove these lines.